app/models/course/discussion/topic.rb

```diff
@@ -13,4 +13,12 @@ class Course::Discussion::Topic < ActiveRecord::Base
   def subscribed_by?(user)
     subscriptions.where(user: user).any?
   end
+
+  # Create subscription for a user
+  #
+  # @param [User] user The user who needs to subscribe to this topic
+  def subscribed_by(user)
```

Comment:
the unique index only ensures either a) you get a RecordNotUniqueError or b) you get a validation error. You don't use `save!` so you will ignore the validation error (what if it is failing in other reasons?) but you still get the RecordNotUnique exception.